usr/src/lib/brand/bhyve/zone/bhhwcompat.c

```diff
@@ -72,6 +73,15 @@ main(int argc, char *argv[])
 	}
 
 	emsg[0] = '\0';
+	version = ioctl(fd, VMM_INTERFACE_VERSION, 0);
+	if (version != VMM_CURRENT_INTERFACE_VERSION) {
+		if (verbose) {
+			fprintf(stderr, "mismatched interface version, "
+			    "expecting %d, got %d\n",
+			    VMM_CURRENT_INTERFACE_VERSION, version);
+		}
+		exit(1);
+	}
```

Comment:
So, a couple of things.

1. What circumstances would lead this to be different? This binary itself is not actually using that interface...does it matter that it's different?
2. This program is used by `sysinfo` to populate the `Bhyve Capable` key. If this is throwing a message to stderr, that's going to show up when people run `sysinfo`. I'm not sure that's something we actually want to do.

Reply:
1.) Mismatched bhhwcompat source vs. kernel.  It DOES use the interface, but only to determine the if this HW can support BHYVE.  It really shouldn't happen given we build the whole world at once, and this program is in illumos-joyent along with the VMM_CURRENT_INTERFACE_VERSION.  Think of it as an overwrought ASSERT().

2.) I'm following the prior art here (see below the block under review) in sending error messages to stderr before exit(1).